org.eclipse.ice.commands/src/main/java/org/eclipse/ice/commands/ConnectionManager.java

```diff
@@ -104,34 +111,37 @@ public Connection openConnection(ConnectionConfiguration config) throws JSchExce
 
 			// Try go get and open the new session
 			try {
-				newConnection.setSession(newConnection.getJShellSession().getSession(username, hostname));
-			} catch (JSchException e) {
+				ClientSession session = client.connect(username, hostname, SshConstants.DEFAULT_PORT).verify(DEFAULT_TIMEOUT).getSession();
+				newConnection.setSession(session);
+			} catch (IOException e) {
 				logger.error("Couldn't open session with given username and hostname. Exiting.", e);
-				throw new JSchException();
+				throw e;
 			}
 
-			// Authorize the JSch session with a ConnectionAuthorizationHandler
-			authorizeSession(newConnection);
-
+			// I don't believe this is required for Mina-ssh
+			//
 			// JSch default requests ssh-rsa host checking, but some keys
 			// request other types. Loop through the available authorization types
 			// and add them to the session.
-			for(String type : authTypes) {
-				newConnection.getSession().setConfig("server_host_key", type);
+//			for(String type : authTypes) {
+//				newConnection.getSession().setConfig("server_host_key", type);
+//			}
+
+			// Set the delegate to reject all, then set the key verifier to check the strict option
+			ServerKeyVerifier delegate = RejectAllServerKeyVerifier.INSTANCE;
+			if (knownHosts != null) {
+				newConnection.getSession().setServerKeyVerifier(new DefaultKnownHostsServerKeyVerifier(delegate, requireStrictHostKeyChecking, Paths.get(knownHosts)));
+			} else {
+				newConnection.getSession().setServerKeyVerifier(new DefaultKnownHostsServerKeyVerifier(delegate, requireStrictHostKeyChecking));
 			}
 
-			// If the user wants to disable StrictHostKeyChecking, add it to the
-			// session configuration
-			if (!requireStrictHostKeyChecking)
```

Comment:
This was something that Caleb specifically requested we leave in, in the event that he wants to use it at some point. Is there a reason this was taken out, or does mina handle hostkey checking differently?

Reply:
It's completely different in Mina, but it should be being handled the same way in this code:

```
ServerKeyVerifier delegate = RejectAllServerKeyVerifier.INSTANCE;
if (knownHosts != null) {
	newConnection.getSession().setServerKeyVerifier(new DefaultKnownHostsServerKeyVerifier(delegate, requireStrictHostKeyChecking, Paths.get(knownHosts)));
} else {
	newConnection.getSession().setServerKeyVerifier(new DefaultKnownHostsServerKeyVerifier(delegate, requireStrictHostKeyChecking));
}
```

I'm not sure there is a test specifically for this though.

Reply:
Ah sorry, this was my mistake. I just completely misread the code - you are right.